README.md

```diff
@@ -137,8 +137,30 @@ not limited to:
 
 * Sending a message with a `null` value for a mandatory field.
 
-The `ProtocolError` message must *not* be used to report Sass errors or errors
-running custom functions or importers.
+The `ProtocolError` message may be used to report errors for violating any
+additional requirements related to this protocol set out by the Sass language.
+This includes, but is not limited to:
+
+* Sending an invalid `Value` that violates requirements of a SassScript value.
+
+The `ProtocolError` message must *not* be used to report any recoverable errors.
+This includes, but is not limited to: 
+
+* The host must send a response with a result of `error` for:
+
+  * Errors running custom functions
+
+  * Errors running importers
+
+* The compiler must send a `CompileFailure` for:
+
+  * Errors in responses received from the host
+
+  * Errors parsing custom function signatures in `CompileRequest`
+
+  * Errors parsing URL in `ImporterResponse` or `FileImporterReponse`
+
+  * Sass errors
```

Comment:
Can we include these inline in the places where the protocol documentation says "must not"?